### DISCUSSION TITLE
Migration to Improved meeting documentation

### DISCUSSION THREAD

**Discussion by andriy-andreyev** [webex/webex-js-sdk#3608] [General]
Hi! Currently I try to learn changes in JS SDK 3.0 and faced with the problem that documentation is not clear enough and I cannot create a call with new SDK.
I'm follow https://github.com/webex/webex-js-sdk/wiki/Migration-to-improved-meetings-associated-with-a-space#sdk-changes
andI have a couple questions.

1. What is the way to automate Service App Guest token pair creating? I found this question https://community.cisco.com/t5/webex-for-developers/service-app-retrieve-the-access-and-refresh-the-token-pair/m-p/4930107

"Yes, in case of Service App - for the first attempt, you need to generate the Access Token & Refresh Token pair from the Service App settings page in Webex Developer Portal only. Because in Service App, you don't have the place to feed redirect URLs or, Code. Even to generate the Code - you need OAuth URL which is not available from the Service App details page.
But straight from the next time, since you already have the Refresh Token available - for generating a new Access Token programmatically using that Refresh Token the logic/flow is the same between an Integration and Service App."

Is it true? I can automate only updating toke using refresh one? Refresh token expires in 90 days. So, if I won't update token after 90 days my deployment will not work and it is needed to go to My Application tab and recreate token pair manually? It looks strange, IMHO. If I have many deployments I need to take care about tokens and control it manually or create additional soft to do it. It is very inconveniently. I belive I'm wrong.

2. Meeting creation. How can I call to another participant if I already have access token?
Before, I did something like:
webex.rooms.create(...);
webex.meetings.create(room.id);
meeting.join();
activeMeeting.invite({ emailAddress: calleeEmail });

But what I should do for now?
From documentation:
"Create a meeting by calling the API https://webexapis.com/v1/meetings and obtain a meetingId or sipUrl.
Find the full details of the API here
Read through Integrations and Service Apps to use appropriate access tokens.
Check if the improved meeting experience is enabled by checking the configuration of your webex object - webex.meetings.config.experimental.enableUnifiedMeetings. Toggle the improved meeting experience by setting the above flag to true if not already enabled
Creating a Webex meeting using either a meeting ID or a SIP URL:
webex.meetings.create('meeting-id or sipUrl').then((meeting) => {
  console.log(meeting);
});"

As far as I understand, I need call API https://webexapis.com/v1/meetings to create meeting-id. Then I use it here
webex.meetings.create('meeting-id or sipUrl').then((meeting) => {
  console.log(meeting);
});
So, I need to mix REST API calls and JS SDK using at the same time, I'm not? It is not looks like the most comfortable SDK experience.
But OK, I'm follow doc:
This doc https://developer.webex.com/docs/api/v1/meetings/create-a-meeting describe how to create adhoc meeting using rooms that is deprecated for now. Looks like it is not updated yet or I did't understand something, please advice.

Resume. For my product I used Integrations and Guest Issuers Webex application. I make a call from Guest to Integration using the Integration's email. Now it should be Integration and Service App Guest, I guess. Can you please share the straight way (documentation from scratch step by step) how can I create a call between Integration and Service App Guest?

**Comment by sreenara** [comment]
@andriy-andreyev thanks for your questions. We will respond to you shortly!

**Comment by mkesavan13** [comment]
Hi @andriy-andreyev 

Thanks for posting this question. Let me try and explain things better,

When the doc says,

> The support for direct use of roomId/spaceId in Webex JS SDK and Meeting Widgets for creation and joining a meeting is being deprecated.

It means that the creation of an Adhoc meeting by using the `roomId` is deprecated in the V3 SDK (or to be precise, from the `webex@2.60.0` SDK version). However, the Adhoc meeting creation using the [/api/v1/meetings/create-a-meeting](https://developer.webex.com/docs/api/v1/meetings/create-a-meeting) is not deprecated. So, that document is intact. 

Let me try to list down the steps here again,

### Creating a Meeting using Room ID
1. The `webex.meetings.create(roomId)` that used to work earlier won't work anymore
2. The new flow is as follows,
    1. Set the `roomId` to your room ID and `adhoc` to `true` with this [Create a Meeting API](https://developer.webex.com/docs/api/v1/meetings/create-a-meeting)
    2. That will Create a meeting in the Space and you will get a response that contains meeting information
    3. This meeting information will have a `sipAddress` that can be passed on to the SDK API `webex.meetings.create(sipAddress)`
    4. After this point, the SDK will function as usual

I will post about the Service Apps and Guest Token soon in another comment
